lib/fluent/plugin/out_kafka.rb

```diff
@@ -162,7 +162,7 @@ def emit(tag, es, chain)
         value = @formatter_proc.call(tag, time, record)
 
         log.on_trace { log.trace("message send to #{topic} with key: #{partition_key} and value: #{value}.") }
-        producer.produce(value, topic: topic, partition_key: partition_key)
+        producer.produce(value, topic: topic, key: partition_key)
```

Comment:
partition_key is for partition, not message key.
Does this change, using partition_key for message key, realize your purpose?